doc/source/user_guide/groupby.rst

```diff
@@ -1049,16 +1049,14 @@ for both ``aggregate`` and ``transform`` in many standard use cases. However,
 
 The dimension of the returned result can also change:
 
-.. ipython::
-
-    In [8]: grouped = df.groupby('A')['C']
+.. ipython:: python
 
-    In [10]: def f(group):
-       ....:     return pd.DataFrame({'original': group,
-       ....:                          'demeaned': group - group.mean()})
-       ....:
+    grouped = df.groupby('A', group_keys=False)['C']
 
-    In [11]: grouped.apply(f)
+    def f(group):
+        return pd.DataFrame({'original': group,
+                             'demeaned': group - group.mean()})
+    grouped.apply(f)
```

Comment:
Experimenting with this example, I noticed that if you add a `reset_index(drop=True)` to the dataframe construction inside the function (so it no longer preserves the index of the original input), this example breaks with this PR (but worked before):

```
In [14]: def f(group):
    ...:     return pd.DataFrame({'original': group,
    ...:                          'demeaned': group - group.mean()}).reset_index(drop=False)

In [15]: df.groupby('A')['C'].apply(f)
/home/joris/miniconda3/envs/dev/bin/ipython:1: FutureWarning: Not prepending group keys ...
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
...
ValueError: cannot reindex from a duplicate axis
```

while on released version this works:

```
In [3]: df.groupby('A')['C'].apply(f)
Out[3]: 
       original  demeaned
A                        
bar 0 -1.270588 -0.638954
    1  0.247197  0.878831
    2 -0.871512 -0.239878
foo 0 -0.506580 -0.720132
    1  0.473710  0.260158
    2 -0.417296 -0.630848
    3 -0.249227 -0.462779
    4  1.767152  1.553600
```

Reply:
Thanks! This pointed to a bug in `_is_indexed_like`. Fixed and test added.